addons/binding/org.openhab.binding.hyperion/src/main/java/org/openhab/binding/hyperion/internal/HyperionDiscoveryParticipant.java

```diff
@@ -58,6 +58,10 @@ public DiscoveryResult createResult(ServiceInfo service) {
 
         String longName = service.getName();
         int pos = longName.indexOf("@");
+        if (pos < 0 || pos >= longName.length()) {
```

Comment:
`indexOf` will either return a valid position within the string bounds or `-1` so you can simply check `if (pos < 0) { return null; }`